pkg/controller/schedule_controller.go

```diff
@@ -176,35 +181,61 @@ func parseCronSchedule(itm *velerov1.Schedule, logger logrus.FieldLogger) (cron.
 	return schedule, nil
 }
 
-func (c *scheduleReconciler) submitBackupIfDue(ctx context.Context, item *velerov1.Schedule, cronSchedule cron.Schedule) error {
-	var (
-		now                = c.clock.Now()
-		isDue, nextRunTime = getNextRunTime(item, cronSchedule, now)
-		log                = c.logger.WithField("schedule", kubeutil.NamespaceAndName(item))
-	)
+// checkIfBackupInNewOrProgress check whether there are backups created by this schedule still in New or InProgress state,
+func (c *scheduleReconciler) checkIfBackupInNewOrProgress(schedule *velerov1.Schedule) bool {
+	log := c.logger.WithField("schedule", kubeutil.NamespaceAndName(schedule))
+	backupList := &velerov1.BackupList{}
+	options := &client.ListOptions{
+		Namespace: schedule.Namespace,
+		LabelSelector: labels.Set(map[string]string{
+			velerov1.ScheduleNameLabel: schedule.Name,
+		}).AsSelector(),
+	}
+
+	err := c.List(context.Background(), backupList, options)
+	if err != nil {
+		log.Errorf("fail to list backup for schedule %s/%s: %s", schedule.Namespace, schedule.Name, err.Error())
+		return true
+	}
+
+	for _, backup := range backupList.Items {
+		if backup.Status.Phase == velerov1.BackupPhaseNew || backup.Status.Phase == velerov1.BackupPhaseInProgress {
+			return true
+		}
+	}
+
+	return false
+}
+
+// ifDue check whether schedule is due to create a new backup.
+func (c *scheduleReconciler) ifDue(schedule *velerov1.Schedule, cronSchedule cron.Schedule) bool {
+	isDue, nextRunTime := getNextRunTime(schedule, cronSchedule, c.clock.Now())
+	log := c.logger.WithField("schedule", kubeutil.NamespaceAndName(schedule))
 
 	if !isDue {
 		log.WithField("nextRunTime", nextRunTime).Debug("Schedule is not due, skipping")
-		return nil
+		return false
 	}
 
+	log.WithField("nextRunTime", nextRunTime).Info("Schedule is due, going to submit backup.")
```

Comment:
Please remove the `going to submit backup` part or move it into the `submitBackup` function as there is another checking before submitting the backup

Reply:
Removed.